services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/UploadWithKnownContentLengthHelper.java

```diff
@@ -241,18 +329,41 @@ public void onError(Throwable t) {
         public void onComplete() {
             log.debug(() -> "Received onComplete()");
             isDone = true;
-            completeMultipartUploadIfFinish(asyncRequestBodyInFlight.get());
+            if (!isPaused) {
+                completeMultipartUploadIfFinish(asyncRequestBodyInFlight.get());
```

Comment:
This is an interesting edge case... We should talk about it.

Reply:
Yeah do we want to pause when all uploadParts() are completed, and the CompleteMPU call is sent but not finished?
or always let CompleteMPU finish?